core/dbt/env_vars.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+from dbt.cli import params
+from dbt.deprecations import warn
+from dbt_common.constants import ENGINE_ENV_PREFIX
+from dbt_common.context import get_invocation_context
+
+# These are env vars that are not in the params module, but are still allowed to be set.
+# New additions to this list should use the new naming scheme, unless they are being added because
+# they already existed, but we didn't know about them previously.
+# TODO: Should at least some of these become (undocumented) cli param options?
+_ADDITIONAL_ENGINE_ENV_VARS: List[str] = [
+    "DBT_INVOCATION_ENV",
+    "DBT_RECORDED_FILE_PATH",
+    "DBT_TEST_STATE_MODIFIED",  # TODO: This is testing related, should we do this differently?
+    "DBT_PACKAGE_HUB_URL",
+    "DBT_DOWNLOAD_DIR",
+    "DBT_PP_FILE_DIFF_TEST",  # TODO: This is testing related, should we do this differently?
+    "DBT_PP_TEST",  # TODO: This is testing related, should we do this differently?
+]
+
+
+@dataclass(frozen=True)
+class EngineEnvVar:
+    name: str
+    old_name: Optional[str] = None
+
+
+def _create_engine_env_var(name: str) -> EngineEnvVar:
```

Comment:
nit: perhaps this would be better-suited as a post_init method on EngineEnvVar

Reply:
I think I'm actually gonna make this one an `__init__` instead of `__post_init__`. The reason being we're partially determining whether the passed in argument should be the `name` or the `old_name`, and that feels more `__init__`-y